src/segment-loader.js

```diff
@@ -1145,10 +1145,13 @@ export default class SegmentLoader extends videojs.EventTarget {
 
   handleCaptions_(simpleSegment, captions) {
     if (this.checkForAbort_(simpleSegment.requestId) ||
-      this.abortRequestEarly_(simpleSegment.stats) ||
-      // This could only happen with fmp4 segments, but
-      // should still not happen in general
-      captions.length === 0) {
+      this.abortRequestEarly_(simpleSegment.stats)) {
+      return;
+
+    // This could only happen with fmp4 segments, but
+    // should still not happen in general
+    } else if (captions.length === 0) {
```

Comment:
Minor, but for readability I find it's easier when this starts its own condition, rather than being part of an else if (similar to how the rest of the code doesn't reside in an else since the first condition returns).